probe/endpoint/nat.go

```diff
@@ -50,7 +50,10 @@ func toMapping(f Flow) *endpointMapping {
 // ApplyNAT duplicates Nodes in the endpoint topology of a
 // report, based on the NAT table as returns by natTable.
 func (n NATMapper) ApplyNAT(rpt report.Report, scope string) {
-	n.WalkFlows(func(f Flow) {
+	if n.Conntracker == nil {
+		return
+	}
```

Comment:
>  It feels like the right way to do this isn't to make nil checks deep in component logic,

It anything this is pushing the nil check further in to the logic.

And in either case the nil check is virtually the same place (it was before the call to ApplyNat, now its the first thing that happens ~after the call).

If you want to provide a noop implementation, then I'm fine with that.  But that would entail making NATMapper an interface, which can be nil, so kinda defeats the whole purpose.